corehq/apps/userreports/pillow.py

```diff
@@ -415,14 +415,16 @@ def process_change(self, change):
             eval_context = EvaluationContext(doc)
             # make copy to avoid modifying list during iteration
             adapters = list(self.table_adapters_by_domain[domain])
+            doc_subtype = change.metadata.document_subtype
             for table in adapters:
                 if table.config.filter(doc, eval_context):
                     if table.run_asynchronous:
                         async_tables.append(table.config._id)
                     else:
                         self._save_doc_to_table(domain, table, doc, eval_context)
                         eval_context.reset_iteration()
-                elif table.config.deleted_filter(doc) or table.doc_exists(doc):
+                elif (doc_subtype is None
```

Comment:
essentially the same change as https://github.com/dimagi/commcare-hq/pull/24620, but in `process_change` instead of hte bulk processor